build-system/tasks/make-extension/index.js

```diff
@@ -258,32 +266,91 @@ async function makeExtensionFromTemplates(
 
   log(`${blurb.join('\n\n')}\n`);
 
-  return true;
+  return writtenFiles;
+}
+
+/**
+ * @param {Array<string>} paths
+ * @param {function(...*):*} fn
+ * @return {Promise}
+ */
+async function affectsExistingPaths(paths, fn) {
+  const stashStdout = getStdout(`git stash push --keep-index`);
+
+  await fn();
+
+  const head = getStdout('git rev-parse HEAD').trim();
+  getOutput(`git checkout ${head} ${paths.join(' ')}`);
+
+  if (!stashStdout.startsWith('No local changes')) {
+    getOutput('git stash pop');
+  }
+}
+
+/**
+ * Generates and tests an extension.
+ * (Unit tests for the generator are located in test/test.js)
+ *
+ * @return {!Promise}
+ */
+async function runMakeExtensionTests() {
+  let error;
+
+  await affectsExistingPaths([extensionBundlesJson], async () => {
+    const name = 'amp-generated-for-test';
+
+    const writtenFiles = [
+      ...(await makeExtensionFromTemplates(TEMPLATES_CLASSIC, '.', {
+        name,
+        version: '0.1',
+      })),
+      ...(await makeExtensionFromTemplates(TEMPLATES_BENTO, '.', {
+        name,
+        version: '1.0',
+        bento: true,
+      })),
+    ];
+
+    for (const command of [
+      `amp build --extensions=${name} --core_runtime_only`,
+      `amp unit --nohelp --headless --files="extensions/${name}/**/test/test-*.js"`,
+    ]) {
+      log(cyan(command));
```

Comment:
```suggestion
      log('Running', cyan(command) + '...');
```

Reply:
I find that the trailing `...` makes the command hard to copy, so I did this instead:
```suggestion
    log('Running...');
    log(cyan(command));
```

Reply:
Unfortunately, this breaks the convention used everywhere else in `build-system/` and makes the CI logs longer than they need to be, so I think there's value in logging on a single line. Also, curious about why this needs to be copied from the CI logs, which already contain other strings like timestamps.

Reply:
Ok, changed to your original suggestion. 

I'm logging the command for visibility, it's not printed by `getOutput`.